src/events.rs

```diff
@@ -115,6 +115,9 @@ pub enum WindowEvent {
     ///
     /// For more information about DPI in general, see the [`dpi`](dpi/index.html) module.
     HiDpiFactorChanged(f64),
+
+    /// The window has moved to another monitor
+    MonitorChanged(Option<String>),
```

Comment:
Two points:
- The `MonitorChanged` event should contain a [`MonitorId`](https://docs.rs/winit/0.18.1/winit/struct.MonitorId.html), not a `String`.
- Why does `MonitorChanged` contain an `Option` instead of a raw value? Your code never emits a `None`,

Reply:
> Two points:
> 
> * The `MonitorChanged` event should contain a [`MonitorId`]
(https://docs.rs/winit/0.18.1/winit/struct.MonitorId.html), not a `String`.

This is currently not possible because MonitorId does not implement `PartialEq`. The main reason that this is not derivable is that [`sctk::output::OutputMgr`](https://docs.rs/smithay-client-toolkit/0.4.4/smithay_client_toolkit/output/struct.OutputMgr.html) does not implement PartialEq, too.

> * Why does `MonitorChanged` contain an `Option` instead of a raw value? Your code never emits a `None`,

I will go for a `String` for now, you're right.

Reply:
You should be able to implement `PartialEq` for `MonitorId` by comparing the backend's monitor ID - it looks like every platform has some method for retrieving uniquely identifying information. For Wayland, that would be `MonitorHandle::get_native_identifier`.